src/test/java/software/amazon/awssdk/crt/test/CustomKeyOpsTest.java

```diff
@@ -131,31 +129,109 @@ public void performOperation(TlsKeyOperation operation) {
+        Assume.assumeTrue(AWS_TEST_MQTT311_CUSTOM_KEY_OPS_CERT != null);
+        Assume.assumeTrue(AWS_TEST_MQTT311_CUSTOM_KEY_OPS_KEY != null);
 
-        TestKeyOperationHandler myKeyOperationHandler = new TestKeyOperationHandler(TEST_PRIVATEKEY, false, false);
+        TestKeyOperationHandler myKeyOperationHandler = new TestKeyOperationHandler(AWS_TEST_MQTT311_CUSTOM_KEY_OPS_KEY, false, false);
         TlsContextCustomKeyOperationOptions keyOperationOptions = new TlsContextCustomKeyOperationOptions(myKeyOperationHandler);
-        try {
-            connect(keyOperationOptions);
+        keyOperationOptions.withCertificateFilePath(AWS_TEST_MQTT311_CUSTOM_KEY_OPS_CERT);
+
+        try (TlsContextOptions contextOptions = TlsContextOptions.createWithMtlsCustomKeyOperations(keyOperationOptions);
+                TlsContext context = new TlsContext(contextOptions);)
+            {
+                connectDirectWithConfig(
+                    context,
+                    AWS_TEST_MQTT311_IOT_CORE_HOST,
+                    8883,
+                    null,
+                    null,
+                    null);
+                disconnect();
+                close();
+            }
+    }
+
+    // Not ideal, but I don't see the point of making a new file just for a single MQTT5 Custom Key Ops test.
+    static final class LifecycleEvents_Futured implements LifecycleEvents {
+        CompletableFuture<Void> connectedFuture = new CompletableFuture<>();
+        @Override
+        public void onAttemptingConnect(Mqtt5Client client, software.amazon.awssdk.crt.mqtt5.OnAttemptingConnectReturn onAttemptingConnectReturn) {}
+        @Override
+        public void onConnectionSuccess(Mqtt5Client client, software.amazon.awssdk.crt.mqtt5.OnConnectionSuccessReturn onConnectionSuccessReturn) {
+            connectedFuture.complete(null);
         }
-        catch (Exception ex) {
-            fail("Exception during connect: " + ex.toString());
+        @Override
+        public void onConnectionFailure(Mqtt5Client client, software.amazon.awssdk.crt.mqtt5.OnConnectionFailureReturn onConnectionFailureReturn) {
+            connectedFuture.completeExceptionally(new Exception("Could not connect!"));
         }
-        disconnect();
-        close();
+        @Override
+        public void onDisconnection(Mqtt5Client client, software.amazon.awssdk.crt.mqtt5.OnDisconnectionReturn onDisconnectionReturn) {}
+        @Override
+        public void onStopped(Mqtt5Client client, software.amazon.awssdk.crt.mqtt5.OnStoppedReturn onStoppedReturn) {}
     }
 
     @Test
-    public void testExceptionFailurePath() {
+    public void testHappyPathMQTT5() {
         skipIfNetworkUnavailable();
-        Assume.assumeTrue(TEST_PRIVATEKEY != null && TEST_PRIVATEKEY != "");
+        // Not ideal, but I don't see the point of making a new file just for a single MQTT5 Custom Key Ops test.
+        String AWS_TEST_MQTT5_IOT_CORE_HOST = System.getenv("AWS_TEST_MQTT5_IOT_CORE_HOST");
```

Comment:
Is there a reason why these strings can't be added to MqttClientConnectionFixture.java like the rest?